File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cs=0
        ms=float('-inf')
        for i in range(len(nums)):
            cs+=nums[i]
            if(cs>ms):
                ms=cs
            if(cs<0):
                cs=0
        return ms

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxi=Integer.MIN_VALUE;
        int s=0;
        
        for(int i=0;i<nums.length;i++){
            s+=nums[i];
            
            maxi=Math.max(maxi,s);
            if(s<0){
                s=0;
            }
        }
        return maxi;
    }
}


     